nemo/collections/asr/parts/utils/transcribe_utils.py

```diff
@@ -282,14 +284,26 @@ def prepare_audio_data(cfg: DictConfig) -> Tuple[List[str], bool]:
             logging.error(f"The input dataset_manifest {cfg.dataset_manifest} is empty. Exiting!")
             return None
 
-        all_entries_have_offset_and_duration = True
-        for item in read_and_maybe_sort_manifest(cfg.dataset_manifest, try_sort=cfg.presort_manifest):
-            if not ("offset" in item and "duration" in item):
-                all_entries_have_offset_and_duration = False
-            audio_key = cfg.get('audio_key', 'audio_filepath')
-            audio_file = get_full_path(audio_file=item[audio_key], manifest_file=cfg.dataset_manifest)
-            filepaths.append(audio_file)
+        audio_key = cfg.get('audio_key', 'audio_filepath')
+
+        with NamedTemporaryFile("w", suffix=".json", delete=False) as durations_f, open(cfg.dataset_manifest, "rt") as f:
+            for item in f:
+                item = json.loads(item)
+                item["audio_filepath"] = get_full_path(item["audio_filepath"], cfg.dataset_manifest)
+                if item.get("duration") is None:
+                    sound = sf.SoundFile(item["audio_filepath"])
+                    duration = sound.frames / sound.samplerate
+                    item["duration"] = duration
+                print(json.dumps(item), file=durations_f, flush=True)
```

Comment:
why print is required here?

Reply:
It's not. I coped it from: https://github.com/NVIDIA/NeMo/pull/9198/files/9a928206ec297f0b2652f4c382883c3bb29fd8a3#diff-f7b768bed0ebc1145787134787f17cc8dfb0defbf8758c8cb759b96e39fbbec5R361

In fact, there is a subtle bug with using print(). Flushing at a new line is not guaranteed. So the code I linked to actually chops off the last few lines of a manifest file. That was one of the reasons why I took a while to get this PR up. FYI @pzelasko , since I think you worked on pre-sorting before.

I will change from using print() to simply do json.write or json.save or w/e as an update to this PR.

https://docs.python.org/3/library/functions.html#print

Reply:
Indeed my bad; I forgot to add `f.flush()` at the end of line writing. `f.write()` would have the same issue. `print` is just fine IMO but if you want to change it I don't mind.